src/logic.pl

```diff
@@ -1,23 +1,148 @@
-        Pos3 is Pos1 +1,
-        populate(Pos3,Pos2).
+%.
+
+add_piece(_,0,_,_).
+
+add_piece(Piece,Size,Direction, Position):-
+    (Direction = l;
+        Direction = r),
+    asserta(piece_position(Piece, h,Position)),
+    next_position(Position,Direction,Next_position),
+    Size2 is Size -1,
+    add_piece(Piece,Size2,Direction, Next_position).
+
+add_piece(Piece,Size,Direction, Position):-
+    (Direction = u;
+        Direction = d),
+    asserta(piece_position(Piece, v,Position)),
+    next_position(Position,Direction,Next_position),
+    Size2 is Size -1,
+    add_piece(Piece,Size2,Direction, Next_position).
+
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%%%%%%%%%%%%%%%%%% 2nd PHASE OF THE GAME %%%%%%%%%%%%%%%%%%%
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+
+% Move in second phase. Checks if it's possible to remove a piece from the board, and if so, removes it. After that, calculates the points and finishes the move.
+move2P(Player, Piece_id, SCB, SCW) :-
+    findall(Position, piece_position(Piece_id, _, Position),  Positions),
+    piece_position(Piece_id, Direction, _),
+    check_move(Player, Piece_id, Positions, SCB, SCW),
+    remove_piece(Piece_id),
+    calculate_points(Player, Piece_id, Direction, Positions, SCB, SCW, Points),
+    finish_move(Player, SCB, SCW, Piece_id, Points).
+
+% Checks if the move is valid
+check_move(Player, Piece_id, Positions, SCB, SCW) :-
+    piece_owner(Player, Piece_id), %Checks if the piece belongs to the player
+    check_valid_remove(Positions, SCB, SCW).    %Checks if there is a Score Counter in the piece's positions
+
+check_valid_remove([], _, _).
+check_valid_remove([Position|T], SCB, SCW) :-
+    check_valid_remove(T, SCB, SCW),
+    Position \= SCB,
+    Position \= SCW.
+
+% removes the piece from the board
+remove_piece(Piece_id) :-
+    retractall(piece_position(Piece_id, _, _)).
+
+% Calculates the points of the move
+calculate_points(Player, Piece_id, Direction, Positions, SCB, SCW, Points) :-
+    get_line_values(Direction, Positions, Values),
+    pieces_in_line(Values, Pieces),
+    sc_in_line(Values, SCB, SCW, SC),
+    piece_value(Piece_id, Value),
+    Points is (Pieces * Value)*SC.
+
+% Gets the values of the line
+get_line_values(Direction, [Position|T], Values) :-
+    Direction is 'H',
+    Line is Position div 10,
+    findall(V, between(0+(1*Line), 9+(1*Line), V), Values).
+
+get_line_values(Direction, [Position|T], Values) :-
+    Direction is 'V',
+    Column is Position mod 10,
+    append([], [Column, Column+10, Column+20, Column+30, Column+40, Column+50, Column+60, Column+70, Column+80, Column+90], Values).
+
+
+% Checks the number of pieces in the line
+pieces_in_line([], 0).
+pieces_in_line([Value|T], Pieces) :-
+    piece_position(_, _, Value),
+    Pieces is Pieces1 + 1,
+    pieces_in_line(T, Pieces1).
+    
+pieces_in_line([Value|T], Pieces) :-
+    pieces_in_line(T, Pieces).
```

Comment:
There is a problem, you are incrementing every time you found a piece, however, if a piece has size 3, has the same direction as the piece removed, and is in the same line as the piece removed then you are going to increment 3 because the piece occupies 3 positions, maybe we should have an extra list or predicate where we insert the id of the pieces found and check if the price in the position is a new one or one that has already been added, we can also try to use the setof() the get all the id from the line

Reply:
Adding to my previous comment, we could still use the values and have an auxiliary predicate that checks if a position is in the values, we could then use setof(Id, (piece_position(Id,_,Position), in_list(values, Position)),Res)

Or something like that